paddle/fluid/pybind/imperative.cc

```diff
@@ -766,11 +777,20 @@ void BindImperative(py::module *m_ptr) {
       .def("__setitem__",
            [](std::shared_ptr<imperative::VarBase> &self, py::handle _index,
               py::object &value_obj) {
+             VLOG(4) << "Call __setitem__";
              auto self_tensor =
                  self->MutableVar()->GetMutable<framework::LoDTensor>();
-             PyObject *index_ptr = !PyTuple_Check(_index.ptr())
-                                       ? PyTuple_Pack(1, _index.ptr())
-                                       : _index.ptr();
+             PyObject *index_ptr;
+             // NOTE(zhiqiu): PyTuple_Pack increases refcount while PyTuple_New
+             // + PyTuple_SetItem not.
+             // see details:
+             // https://stackoverflow.com/questions/8528483/reference-counts-in-a-python-c-extension
+             if (PyTuple_Check(_index.ptr())) {
+               index_ptr = _index.ptr();
+             } else {
+               index_ptr = PyTuple_New(1);
```

Comment:
I am rather curious that when the `index_ptr` would be released (when the reference count of `index_ptr` decreases to zero)? Is there any risk that `index_ptr` cause memory leak?

Reply:
Yes, it is another problem and I'm working on it .